front/src/applications/operationalStudies/helpers/TrainOpProjectionLazyLoader.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  osrdEditoastApi,
+  type PathProperties,
+  type PostTrainScheduleOccupancyBlocksApiResponse,
+  type PostTrainScheduleProjectPathOpApiResponse,
+} from 'common/api/osrdEditoastApi';
+import type { TimetableItemId } from 'reducers/osrdconf/types';
+import {
+  extractEditoastIdFromPacedTrainId,
+  extractEditoastIdFromTrainScheduleId,
+  formatEditoastIdToTrainScheduleId,
+  isTrainScheduleId,
+} from 'utils/trainId';
+
+import TrainProjectionLazyLoaderAbstract, {
+  type ProjectionResult,
+  type TrainProjectionLazyLoaderOptions,
+} from './TrainProjectionLazyLoaderAbstract';
+
+export default class TrainOpProjectionLazyLoader extends TrainProjectionLazyLoaderAbstract {
+  readonly operationalPoints: PathProperties['operational_points'];
```

Comment:
Instead of storing the operationalPoints here, you can store directly the timetableItemProjectedOpIds and operationalPointDistances
```suggestion
  readonly opIds: string[];
  readonly opDistances: number[];
```

You will have to compute them in the constructor, and so you won't have to compute them at each processBatch

Reply:
Done